python/mxnet/gluon/trainer.py

```diff
@@ -110,16 +110,29 @@ def _init_optimizer(self, optimizer, optimizer_params):
                             for _ in self._contexts]
 
     def _init_kvstore(self):
-        arg_arrays = {param.name: param.data(self._contexts[0]) for param in self._params}
+        arg_arrays = {}
+        contains_sparse = False
+        for param in self._params:
+            arg_arrays[param.name] = param.data(self._contexts[0])
```

Comment:
did you try to do step on parameter with deferred initialization? what message did you get?

Reply:
Yes I tried the language model example. param.data() was called in the previous implementation (line 113)